Bukkit/src/main/java/com/plotsquared/bukkit/BukkitPlatform.java

```diff
@@ -999,6 +997,17 @@ private void runEntityTask() {
         }), TaskTime.seconds(1L));
     }
 
+    private void removeRoadEntity(Entity entity, Iterator<Entity> entityIterator) {
+        RemoveRoadEntityEvent event = eventDispatcher.callRemoveRoadEntity(BukkitAdapter.adapt(entity));
+
+        entityIterator.remove();
```

Comment:
The removal should happen only if `Entity#remove` is called. Also please add braces to the if below.

Reply:
I don't quite understand the not removing from the iterator, but have adjusted it anyway.